misc/python/materialize/mzcompose.py

```diff
@@ -360,6 +361,30 @@ def run(
             stderr=subprocess.PIPE if capture else 2,
         )
 
+    def inspect_volumes(self, volumes: Container[str]) -> Iterable[Dict[str, Any]]:
+        """
+        Return the JSON from `docker inspect` for each volume with the given compose name
+
+        There is no explicit documentation of the structure of the returned
+        fields, but you can see them in the docker core repo:
+        https://github.com/moby/moby/blob/91dc595e9648318/api/types/types.go#L345-L379
+        """
+        actual_volumes = spawn.capture(
+            ["docker", "volume", "ls", "--format", "{{ .Name }}"], unicode=True
+        ).splitlines()
+        metadata = spawn.capture(
+            ["docker", "inspect", "-f", "{{json .}}", *actual_volumes]
+        )
+        for line in metadata.splitlines():
+            info = json.loads(line)
+            # Labels may be present and null or not present
+            labels = info.get("Labels")
+            if (
+                labels is not None
+                and labels.get("com.docker.compose.volume") in volumes
+            ):
+                yield info
```

Comment:
So, this doesn't seem quite right to me. This finds all volumes created by any composition with the specified name... but that includes volumes created in _other_ compositions. So if passed `mzdata`, this might remove testdrive's mzdata volume as well as the upgrade test's mzdata volume.

Reply:
ah yeah that's a good point.